kubernetes-sigs/cluster-api-provider-azure
# Add support for joining clusters to AKS Fleet

**What type of PR is this?**
/kind feature



**What this PR does / why we need it**:
This PR adds support for joining managed clusters to [AKS Fleet](https://learn.microsoft.com/en-us/azure/kubernetes-fleet/overview) via the field `fleetManager` in the `AzureManagedControlPlane`. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Steps to test:

1. Create a fleet manager in AKS. Microsoft docs: https://learn.microsoft.com/en-us/azure/kubernetes-fleet/quickstart-create-fleet-and-members
2. Add the following field to your `AzureManagedControlPlane` spec
```
fleetManager: 
    group: test-group
    managerName: fleet-manager-name
    managerResourceGroup: fleet-manager-rg
```
3. Create the managed cluster and observe it being added as a member to your fleet manager

- [ ] cherry-pick candidate

**TODOs**:



**Release note**:

```release-note
Add support for joining clusters to AKS Fleet
```
